docs/topics/compatibility-guide-18.md

```diff
@@ -0,0 +1,623 @@
+> - 1.5.20: introduce a warning when non-abstract classes that do not override all abstract members are used
+> - 1.7.0: report a warning if a super call, in fact, accesses an abstract member from a superclass
+> - 1.7.0: report an error in all affected cases if the `-Xjvm-default=all` or `-Xjvm-default=all-compatibility` compatibility modes are enabled;
+>   report an error in the progressive mode
+> - 1.8.0: report an error in cases of declaring a concrete class with a non-overridden abstract method from the superclass, and 
+>   super calls of methods of `Any` overridden as abstract in the superclass
+> - 1.9.0: report an error in all affected cases, including explicit super calls to an abstract method from the super class
+
+
+### Deprecate confusing grammar in `when-with-subject`
+
+> **Issue**: [KT-48385](https://youtrack.jetbrains.com/issue/KT-48385)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.6 deprecated several confusing grammar constructs in `when` condition expressions
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: introduce a deprecation warning on the affected expressions
+> - 1.8.0: raise this warning to an error,
+>   `-XXLanguage:-ProhibitConfusingSyntaxInWhenBranches` can be used to temporarily revert to the pre-1.8 behavior.
+> - \>= 1.9: repurpose some deprecated constructs for new language features
+
+
+### Prevent implicit coercions between different numeric types
+
+> **Issue**: [KT-48645](https://youtrack.jetbrains.com/issue/KT-48645)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: Kotlin will avoid converting numeric values automatically to a primitive numeric type where only a downcast to that type was needed semantically
+>
+> **Deprecation cycle**:
+>
+> - < 1.5.30: the old behavior in all affected cases
+> - 1.5.30: fix the downcast behavior in generated property delegate accessors,
+>   `-Xuse-old-backend` can be used to temporarily revert to the pre-1.5.30 fix behavior
+> - \>= 1.9: fix the downcast behavior in other affected cases
+
+### Make private constructors of sealed classes really private
+
+> **Issue**: [KT-44866](https://youtrack.jetbrains.com/issue/KT-44866)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: After relaxing restrictions on where the inheritors of sealed classes can be declared in the project structure,
+> the default visibility of sealed class constructors became protected. However, until 1.8, Kotlin still allowed calling
+> explicitly declared private constructors a sealed class outside the class scope.
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: report a warning (or an error in the progressive mode) when private constructor of a sealed class 
+>   is called outside of that class
+> - 1.8.0: use default visibility rules for private constructors (call can be resolved to it only if it is inside corresponding class),
```

Comment:
```suggestion
> - 1.8.0: use default visibility rules for private constructors (a call can be resolved to it only if it is inside the corresponding class),
```

Reply:
Maybe better rephrase to avoid "it" ambiguity

Reply:
I've made a proposal; could you please check it?

Reply:
The second it referred to call, that's what was confusing

Reply:
Does the current suggestion suit? It's more wordy but it's more clear I suppose.

Reply:
Again, the second "it" did mean "call". I'll correct and commit the suggestion.

Reply:
```suggestion
> - 1.8.0: use default visibility rules for private constructors (a call to a private constructor can be resolved only if this call is inside the corresponding class),
```

Reply:
Ah, sorry( Thanks!